ego4d/internal/validation/validate.py

```diff
@@ -1478,6 +1478,96 @@ def validate_egoexo_files(
 ) -> List[Error]:
     ret = []
 
+# Pre Survey Checks - Flagging if participant + scenario does not have matching pre_survey
+    dictionary_ = {}
+    for (participant_id, scenario_id, _), p in manifest.participants.items():
+
+        key = (participant_id, scenario_id)
+        if key not in dictionary_:
+            dictionary_[key] = []
+        dictionary_[key].append(p)
+
+    for takes in manifest.takes.values():
+        for take in takes:
+            key = (take.recording_participant_id, take.scenario_id)
+
+            if key not in dictionary_:
+                ret.append(
+                        Error(
+                        ErrorLevel.ERROR,
+                        participant_id,
+                        "participant_scenario_no_pre_survey",
+                        "participant {p.participant_id} + scenario {p.scenario_id} does not have a corresponding pre_survey.",
+                        )
+                    )
+
+# Post Survey Checks - Flagging if participant + take does not have matchin post_survey
+
+    def get_post_survey_paths(captures: List[CaptureMetadataEgoExo]) -> List[ReferencedFile]:
+        dict_post_surveys = {}
+        for capture in captures:
+            if not capture.post_surveys_relative_path.startswith("s3://"):
+                survey_path = os.path.join(capture.university_video_folder_path, capture.post_surveys_relative_path) # make absolute path if not allready
+            else:
+                survey_path = capture.post_surveys_relative_path
+            dict_post_surveys[capture.university_capture_id] = survey_path
+        return dict_post_surveys
+
+    def check_file_post_survey(file_path_dict):
+        for key, value in file_path_dict.items():
+            if not os.path.exists(value):
```

Comment:
Does this work? I think you need to use `pathmgr.exists` since the path starts with `s3://`

Alternatively you can check this in `open_csv` below, which has the advantage of it being in a thread pool for execution. For example:

```python
def open_csv(capture_id_file_path):
    capture_id, file_path = capture_id_file_path
    df = None
    if pathmgr.exists(file_path):
        df = pd.read_csv(pathmgr.open(capture_id_file_path[1]), dtype=object)
    return capture_id_file_path[0], df

# ...

for capture_id, df in tqdm(pool.map(map_fn, vals_to_map), total=len(vals_to_map)):
    if df is None:
        ret.append(<add error for path_does_not_exist here>)
        continue
    dict_pandas_df[capture_id] = df
```